### PR TITLE
Use `strk` as a default fee token

### LINKED ISSUE #2696: Use `strk` as a default `--fee-token`
Currently, user is required to provide `--fee-token` or `--version` everytime they make a transaction.
This is impractical in terms of UX.
Additionally, txs supporting eth fee token are most likely to be removed in the upcoming future https://community.starknet.io/t/snip-16-deprecation-of-transaction-versions-0-1-2/114443

Use `strk` as a default `--fee-token` in sncast operations that require providing it.

Additionally, documentation should be updated to stop showing the `--fee-token` or `--version` argument except for the "Fees and Versions" page.

### PR DESCRIPTION
**Pull request by kkawula**
Closes #2696

## Introduced changes

-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Cast
+
+#### Changed
+
+- `--fee-token` and `--version` flags are now optional, `strk` and `v3` will be inferred automatically
```
Comment (cptartur) [inline CHANGELOG.md:14]:
```suggestion
- `--fee-token` and `--version` flags are now optional, `strk` and `v3` will be used by default
```

File: docs/src/starknet/fees-and-versions.md [hunk 2]
```diff
@@ -10,6 +10,11 @@ This is done using the `--max-gas` and `--max-gas-unit-price` flags.
 Cast allows you to specify either the version of the transaction you want to send or the fee token you want to pay the fees in. This is done using
 the `--version` and `--fee-token` flags.
 
+> ⚠️ **Warning**
+> `--fee-token` flag will be deprecated in the future due to [SNIP-16: Deprecation of Transaction Versions 0,1,2](https://community.starknet.io/t/snip-16-deprecation-of-transaction-versions-0-1-2/114443).
+> It is recommended to use `--version` flag instead or let the version be inferred automatically.
```
Comment (cptartur) [inline docs/src/starknet/fees-and-versions.md:15]: Can we add this as a deprecation warning in the runtime as well? 
Show it when someone passes the `--fee-token` arg but have it display two variants:
* `--fee-token strk` just show warning that `--fee-token` will be removed
* `--fee-token eth` show warning that `--fee-token` will be removed & that eth transactions will not be supported in the future

**Review by cptartur — CHANGES_REQUESTED** [review]

File: docs/src/starknet/fees-and-versions.md (hunk 2 above)
Comment (kkawula) [reply docs/src/starknet/fees-and-versions.md:15]: image
image
without flag:
image

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Requirements validation during `snforge` runtime
 - `snforge check-requirements` command
 
+### Cast
+
+#### Changed
+
+- `--fee-token` and `--version` flags are now optional, `strk` and `v3` will be used by default
+
+### `snforge_scarb_plugin`
```
Comment (ddoktorski) [inline CHANGELOG.md:23]: nit:
```suggestion
```

File: crates/sncast/src/helpers/fee.rs [hunk 4]
```diff
@@ -225,3 +233,29 @@ macro_rules! impl_payable_transaction {
...
+
+fn parse_fee_token(s: &str) -> Result<FeeToken, String> {
+    let deprecation_message = "\x1b[33mwarning:\x1b[0m Specifying '--fee-token' flag is deprecated and will be removed in the future. Use '--version' instead";
+
+    match s.to_lowercase().as_str() {
+        "eth" => {
+            println!("{deprecation_message}");
+            println!("\x1b[33mwarning:\x1b[0m Eth transactions will stop being supported in the future due to 'SNIP-16'");
+        }
+        _ => println!("{deprecation_message}"),
```
Comment (ddoktorski) [inline crates/sncast/src/helpers/fee.rs:257]: Please use [print_as_warning](https://github.com/foundry-rs/starknet-foundry/blob/master/crates/shared/src/print.rs#L4) function

Comment (cptartur) [reply crates/sncast/src/helpers/fee.rs:257]: 👆

File: crates/sncast/src/helpers/fee.rs [hunk 5]
```diff
@@ -225,3 +233,29 @@ macro_rules! impl_payable_transaction {
...
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "eth" => Ok(FeeToken::Eth),
+            "strk" => Ok(FeeToken::Strk),
+            _ => Err(String::from("Possible values: eth, strk")),
+        }
+    }
+}
```
Comment (cptartur) [inline crates/sncast/src/helpers/fee.rs:247]: Why do we need to implement this if the function below is using an already parsed `FeeToken`?

File: docs/src/starknet/fees-and-versions.md [hunk 6]
```diff
@@ -10,6 +10,11 @@ This is done using the `--max-gas` and `--max-gas-unit-price` flags.
 Cast allows you to specify either the version of the transaction you want to send or the fee token you want to pay the fees in. This is done using
 the `--version` and `--fee-token` flags.
 
+> ⚠️ **Warning**
+> `--fee-token` flag will be deprecated in the future due to [SNIP-16: Deprecation of Transaction Versions 0,1,2](https://community.starknet.io/t/snip-16-deprecation-of-transaction-versions-0-1-2/114443).
```
Comment (cptartur) [inline docs/src/starknet/fees-and-versions.md:14]:
```suggestion
> `--fee-token` flag is deprecated and will be removed in the future due to [SNIP-16: Deprecation of Transaction Versions 0,1,2](https://community.starknet.io/t/snip-16-deprecation-of-transaction-versions-0-1-2/114443).
```

**Review by cptartur — COMMENTED** [review]
Can we add a test asserting that both warnings that were added by you display correctly and in correct cases?

File: crates/sncast/src/helpers/fee.rs (hunk 5 above)
Comment (kkawula) [reply crates/sncast/src/helpers/fee.rs:248]: I've updated the logic, It seems to me that FromStr create a clear parsing interface

**Comment by kkawula** [comment]
> Can we add a test asserting that both warnings that were added by you display correctly and in correct cases?

https://github.com/foundry-rs/starknet-foundry/issues/2774

File: CHANGELOG.md [hunk 7]
```diff
@@ -16,7 +16,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 #### Changed
 
- - `snforge_scarb_plugin` will now also emit warnings when errors occur
+- `snforge_scarb_plugin` will now also emit warnings when errors occur
+
+### Cast
+
+#### Changed
```
Comment (kkawula) [inline CHANGELOG.md:23]:
```suggestion
#### Deprecated
```

**Pull request merged**